### ISSUE TITLE
CfnDBCluster: Updating master_username shows replacement of resource while running cdk diff

### ISSUE THREAD

**Issue by fernandesnigel5** [aws/aws-cdk#21919]
### Describe the bug

I am creating a rds Dbcluster resource through cdk. However when i try to update the master_username property and while running the "cdk diff" command it shows that the resource requires replacement. 

However i did go ahead and deployed the updated code anyway, it did update the stack but did not cause any replacement of the resource.

Is there any reason for this glitch or strange behavior?

The cdk code i used to deploy the Dbcluster resource

aurora_cluster = rds.CfnDBCluster(self, 'auroraCluster',
                                          engine='aurora-mysql',
                                          engine_version='5.7.mysql_aurora.2.08.1',
                                          database_name = 'database05sep22',
                                          master_username="masterUser",
                                            master_user_password="masterUserPassword",
                                          )

cdk diff output when i updated the master_username

Resources
[~] AWS::RDS::DBCluster auroraCluster auroraCluster replace
 └─ [~] MasterUsername (requires replacement)
     ├─ [-] masterUser
     └─ [+] masterUser123

### Expected Behavior

As per the cloudformation document for AWS::RDS::DBCluster the property "MasterUsername" says no interruption during update. So it should not be showing as replace while running the cdk diff command.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-masterusername

### Current Behavior

While running the cdk diff command I get resource replace message when i change the master_username property

[~] AWS::RDS::DBCluster auroraCluster auroraCluster replace
 └─ [~] MasterUsername (requires replacement)
     ├─ [-] masterUser
     └─ [+] masterUser123

### Reproduction Steps

It is a python code
=======================
aurora_cluster = rds.CfnDBCluster(self, 'auroraCluster',
                                          engine='aurora-mysql',
                                          engine_version='5.7.mysql_aurora.2.08.1',
                                          database_name = 'database05sep22',
                                          master_username="masterUser123",
                                            master_user_password="masterUserPassword",)
========================

### CDK CLI Version

2.31.0 (build b67950d)

### Node.js Version

8.12.1

### OS

MAC

### Language

Python

**Comment by peterwoodworth** [comment]
I can reproduce the same behavior: that is, running `cdk diff`, getting an expected resource replacement, and then on deploy the resources are not replaced

However, the docs you've linked _do_ specify that changes to this property requires replacement. Additionally, the [CloudFormation resource specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) specifies this property as `immutable`, which is how we know whether to specify resource replacement or not.

So in short, there's nothing on CDK's side of things that can be done here. We need CloudFormation to update the resource specification to make this property defined as `mutable`, then `cdk diff` will no longer show that the cluster needs to be replaced
